### PR TITLE
Quest - sync fails after staying idle in the app for a short period of time

### LINKED ISSUE #1106: Quest - sync fails after staying idle in the app for a short period of time
**Describe the bug**
If the user has been idle for some time in the app, and then tries to synchronize, it fails with a 401 error i.e. unauthorized request. I'm not sure exactly how long this length of time is but it seems somewhere between 5 and 10 minutes of being idle. The user isn't signed out, they are still within the app. So in order to synchronize, the user needs to log out, then log back in again and then synchronize. In addition, the toast message doesn't specify that the user needs to do this, it's something I observed in the logs. The toast message simply says "Sync failed. Please check internet connection or try again later".

**To Reproduce**
1. Log into the Quest app
2. Wait for 10 minutes
3. Go to the settings tab
4. Click sync
5. An error message should appear shortly

**Expected behavior**
I expect to be able to sync as long as I am logged into the app.

**Screenshots**
N/A

**Smartphone (please complete the following information):**
- Device: Samsung A12
- Android version: 11
- Build number: RP1A.200720.012.A127FXXU4AUK1
- Version: 30

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by owais-vd**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #1106 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

**Review by pld — CHANGES_REQUESTED** [review]
can you please add a test that ideally reproduces the error in the issue and is resolved with these change

**Comment by owais-vd** [comment]
> can you please add a test that ideally reproduces the error in the issue and is resolved with these change

done

**Review by pld — APPROVED** [review]
lgtm!

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/register/BaseRegisterActivity.kt [hunk 1]
```diff
@@ -698,6 +695,29 @@ abstract class BaseRegisterActivity :
...
+        is State.Glitch -> state.exceptions
+        is State.Failed -> state.result.exceptions
+        else -> listOf()
+      }
+
+    if (exceptions.map { it.exception }.filterIsInstance<HttpException>().firstOrNull()?.code() ==
+        401
+    ) {
+      showToast(getString(R.string.session_expired))
+      accountAuthenticator.logout()
```
Comment (ekigamba) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/register/BaseRegisterActivity.kt:711]: This is an issue in the current OpenSRP. Hold off on this awaiting discussions within OpenSRP

Comment (ekigamba) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/register/BaseRegisterActivity.kt:711]: See this for more information https://github.com/opensrp/support/issues/108

**Pull request merged**